core/typeSugar.ml

```diff
@@ -2314,56 +2430,10 @@ let make_ft_poly_curry declared_linearity ps effects return_type =
   Types.for_all (ft ps)
 
 (** Make any unannotated parameters monomorphic. *)
-let make_mono pats = List.iter (List.iter (fun (_, _, t) -> Types.Mono.make_type t)) pats;
-
-type usagemap = int stringmap
-let merge_usages (ms:usagemap list) : usagemap =
-      match ms with
-      | [] -> StringMap.empty
-      | (m::ms) -> List.fold_right
-                     (fun m n ->
-                        StringMap.merge
-                          (fun _ xo yo ->
-                             match xo, yo with
-                             | Some x, Some y -> Some (x + y)
-                             | Some x, None   -> Some x
-                             | None, Some y   -> Some y
-                             | None, None     -> None
-                          ) m n) ms m
-
-let uses_of v us =
-  try
-    StringMap.find v us
-  with
-    _ -> 0
-
-let usage_compat =
-  function
-  | [] ->
-    (* HACK: for now we take the conservative choice of assuming that
-       no linear variables are used in empty cases. We could keep
-       track of all variables in scope so that we can treat them as
-       linear if possible. This would require a further map recording
-       all variables that have empty pattern matching 'sink'. *)
-    StringMap.empty
-  | (u::us) ->
-    let same m n =
-      let mvs = List.map fst (StringMap.bindings m) in
-      let vs  = List.append (List.filter (fun v -> not (List.mem v mvs)) (List.map fst (StringMap.bindings n)))
-          mvs in
-      let f v resulting_usages =
-        if StringMap.mem v m && StringMap.mem v n && StringMap.find v m = StringMap.find v n then
-          StringMap.add v (StringMap.find v m) resulting_usages
-        else
-        (* We need to treat anything appearing in this case as unlimited; '2' assures that no
-           matter whether the variable in question is used anywhere else or not, it must be
-           unlimited. *)
-          StringMap.add v 2 resulting_usages in
-      List.fold_right f vs StringMap.empty in
-    List.fold_right same us u
+let make_mono pats = List.iter (List.iter (fun (_, _, t) -> Types.Mono.make_type t)) pats
 
 let usages_cases bs =
-  usage_compat (List.map (fun (_, (_, _, m)) -> m) bs)
+  Usage.align (List.map (fun (_, (_, _, m)) -> m) bs)
```

Comment:
Ah -- I see now -- `align` is `usage_compat`

Reply:
I decided to use the name `align` after I read Garrett's explanation of `usage_compat`.